service/src/main/java/bio/terra/workspace/service/workspace/flight/CheckSpendProfileStep.java

```diff
@@ -45,11 +56,25 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
             .getSpendProfileId()
             .orElseThrow(() -> MissingSpendProfileException.forWorkspace(workspaceUuid));
 
-    SpendProfile spendProfile = spendProfileService.authorizeLinking(spendProfileId, userRequest);
-    if (spendProfile.billingAccountId().isEmpty()) {
-      throw NoBillingAccountException.forSpendProfile(spendProfileId);
+    SpendProfile spendProfile =
+        spendProfileService.authorizeLinking(spendProfileId, bpmEnabled, userRequest);
+
+    if (cloudPlatform == CloudPlatform.GCP) {
```

Comment:
Seems to me these checks should be done when we are minting the spend profile and not here in the step.
This is likely not the only place we will call `authorizeLinking`